### PR TITLE
various compiler warnings

### PR DESCRIPTION
**Pull request by olafhering**

### CODE REVIEW

**Comment by olafhering** [comment]
While you are at it:  The two for loops in dcr_kodak_thumb_load_raw look like python indent, but its C.

**Comment by MartijnKaijser** [comment]
before you start doing a lot of fixes for an ancient version. it seems there's a newer version for cximage 7.02 which may already contain all this.
It has been talked about replacing cximage with something else but still hasn't happened. So maybe upgrading would be better?

**Comment by olafhering** [comment]
Well, not my call. If that goes into master I can see what gcc says. The warnings in Isengard should be fixed either way.

File: lib/cximage-6.0/raw/libdcr.c [hunk 1]
```diff
@@ -2471,8 +2471,12 @@ void DCR_CLASS dcr_sony_decrypt (DCRAW *p, unsigned *data, int len, int start, i
 		for (p->sony_decrypt_p=0; p->sony_decrypt_p < 127; p->sony_decrypt_p++)
 			p->sony_decrypt_pad[p->sony_decrypt_p] = htonl(p->sony_decrypt_pad[p->sony_decrypt_p]);
 	}
-	while (len--)
-		*data++ ^= p->sony_decrypt_pad[p->sony_decrypt_p++ & 127] = p->sony_decrypt_pad[(p->sony_decrypt_p+1) & 127] ^ p->sony_decrypt_pad[(p->sony_decrypt_p+65) & 127];
+	while (len--) {
+		p->sony_decrypt_pad[p->sony_decrypt_p & 127] = p->sony_decrypt_pad[(p->sony_decrypt_p+1) & 127] ^ p->sony_decrypt_pad[(p->sony_decrypt_p+65) & 127];
```
Comment (Memphiz) [inline lib/cximage-6.0/raw/libdcr.c]: shouldn't the last index be + 66 now?

Comment (olafhering) [inline lib/cximage-6.0/raw/libdcr.c]: why? I think the expectation is that the right side is evaluated first, then assigned, and once assigned its incremented.

Comment (Memphiz) [inline lib/cximage-6.0/raw/libdcr.c]: You are right ... it was a bit hard to see and the github page didn't scroll to the right when i thought it should (hence i thought that the p->sony_decrypt_p++ was at the right of the assignment and the +1 of the first index compensated that ;) - all fine as is...

Thats definitly a wtf code line you refactored there ;)

**Comment by Memphiz** [comment]
@MartijnKaijser imo we want this - right?

**Comment by MartijnKaijser** [comment]
If it fixes something, sure. Just wondering if time wouldn't be better spend updating cximage complete and then see what needs fixing. (if this is what you meant)

**Comment by Memphiz** [comment]
Well as we can't force anybody to dig into that mine field - jenkins build and merge ;)

**Pull request merged**